### DISCUSSION TITLE
Custom Offline Dependencies: Enhance Documentation and Rename Var in Source

### DISCUSSION THREAD

**Discussion by danhlee329** [renovatebot/renovate#27644] [Suggest an Idea]
### Tell us more.

This is a follow up to a question asked yesterday (https://github.com/renovatebot/renovate/discussions/27628).  To summarize, there was some confusion and incorrect assumptions made based on what was missing on the documentation as well as from the debug logs that are provided during debugging.

Would be great if we can make some enhancements for better clarification.

- Under [Custom offline dependencies](https://docs.renovatebot.com/modules/datasource/custom/#custom-offline-dependencies), when defining the path to a version file locally via the `file://` prefix, we should note that this version-file is expected to be in the target-Github-repo (or any other source) at the time of cloning (as noted in the discussion: https://github.com/renovatebot/renovate/discussions/27628#discussioncomment-8627792).
- Per question above, I've run into errors from logic in [this function](https://github.com/renovatebot/renovate/blob/1efa2f21da4f4b2c0c984726c1a35038d76732db/lib/util/fs/util.ts#L6-L14) (with `DEBUG` mode on). `baseDir` is confusing because this nomenclature is already used throughout the docs and could assume to mean the `baseDir` that we set via ENV VAR `RENOVATE_BASE_DIR` (it did for me, at least).  Perhaps we can rename this to something different like `refPath` as well as alter the message to say something like `...outside of the target directory`.
  - Example log for context:
```
DEBUG: Preventing access to file outside the base directory (repository=$OWNER/$REPO)
       "path": "$BASE_DIR/module-tracker.json",
       "baseDir": "$BASE_DIR/repos/github/$OWNER/$REPO"
```

**Comment by rarkins** [comment]
Would your requirements be satisfied if we add a configuration option to allow file:// custom datasources to be accessed from anywhere? i.e. no more base directory limitations?

**Reply by danhlee329** [reply]
Yes, that'll be great, I suppose it'll depend on how and where it'll be acceptable to reference the files under Renovate (I.e., limit access within the base Dir or repo, etc), but will help us tremendously.

**Reply by rarkins** [reply]
I think if this option is enabled then it should allow any path to be set, because you trust the repo

**Reply by danhlee329** [reply]
Sounds great, I'll create a feature request around this use case, much appreciated @rarkins!

**Reply by danhlee329** [reply]
Created new discussion https://github.com/renovatebot/renovate/discussions/27675

**Comment by rarkins** [comment]
Created PR to improve the logging so that it doesn't confusingly use the term "base dir": https://github.com/renovatebot/renovate/pull/27659

**Reply by danhlee329** [reply]
Awesome, thanks @rarkins!

**Comment by danhlee329** [comment]
Will continue the discussion in https://github.com/renovatebot/renovate/discussions/27644#discussioncomment-8647429, will close this discussion.
